v1/options.go

```diff
@@ -2,6 +2,166 @@
+//     to use a particular representation for nil, either the `format:emitempty`
+//     or `format:emitnull` field option can be specified.
+//
+//   - In v1, a Go array may be unmarshaled from a JSON array of any length.
+//     In contrast, in v2 a Go array must be unmarshaled from a JSON array
+//     of the same length, otherwise it results in an error.
+//     The [UnmarshalArrayFromAnyLength] option controls this behavior difference.
+//
+//   - In v1, a Go byte array is represented as a JSON array of JSON numbers.
+//     In contrast, in v2 a Go byte array is represented as a Base64-encoded JSON string.
+//     The [FormatBytesWithLegacySemantics] option controls this behavior difference.
+//     To explicitly specify a Go struct field to use a particular representation,
+//     either the `format:array` or `format:base64` field option can be specified.
+//
+//   - In v1, MarshalJSON methods declared on a pointer receiver are only called
+//     if the Go value is addressable. In contrast, in v2 a MarshalJSON method
+//     is always callable regardless of addressability.
+//     The [CallMethodsWithLegacySemantics] option controls this behavior difference.
+//
+//   - In v1, MarshalJSON and UnmarshalJSON methods are never called for Go map keys.
+//     In contrast, in v2 a MarshalJSON or UnmarshalJSON method is eligible for
+//     being called for Go map keys.
+//     The [CallMethodsWithLegacySemantics] option controls this behavior difference.
+//
+//   - In v1, a Go map is marshaled in a deterministic order.
+//     In contrast, in v2 a Go map is marshaled in a non-deterministic order.
+//     The [jsonv2.Deterministic] option controls this behavior difference.
+//
+//   - In v1, JSON strings are encoded with HTML-specific or JavaScript-specific
+//     characters being escaped. In contrast, in v2 JSON strings use the minimal
+//     encoding and only escape if required by the JSON grammar.
+//     The [jsontext.EscapeForHTML] and [jsontext.EscapeForJS] options
+//     control this behavior difference.
+//
+//   - In v1, bytes of invalid UTF-8 within a string are silently replaced with
+//     the Unicode replacement character. In contrast, in v2 the presence of
+//     invalid UTF-8 results in an error. The [jsontext.AllowInvalidUTF8] option
+//     controls this behavior difference.
+//
+//   - In v1, a JSON object with duplicate names is permitted.
+//     In contrast, in v2 a JSON object with duplicate names results in an error.
+//     The [jsontext.AllowDuplicateNames] option controls this behavior difference.
+//
+//   - In v1, when unmarshaling a JSON null into a non-empty Go value it would
+//     inconsistently either zero out the the value or do nothing.
+//     In contrast, in v2 unmarshaling a JSON null would consistently and always
+//     zero out the underlying Go value. The [MergeWithLegacySemantics] option
+//     controls this behavior difference.
```

Comment:
I'm not an expert at english grammar, but the tense here reads a bit inconsistent to me. Should it be "a non-empty Go value it _will_" and "a JSON null _will_ consistently"? The other sentences use present tense ("is permitted" instead of "would be permitted").
```suggestion
//   - In v1, when unmarshaling a JSON null into a non-empty Go value it will
//     inconsistently either zero out the the value or do nothing.
//     In contrast, in v2 unmarshaling a JSON null will consistently and always
//     zero out the underlying Go value. The [MergeWithLegacySemantics] option
//     controls this behavior difference.
```

Reply:
The use of "would" here is the subjunctive mood.

Reply:
Looking at the existing "encoding/json" documentation, it uses language like: "NaN and +/-Inf values **will** return an UnsupportedValueError", so "will" follows existing precedence.